src/components/Jobs/Utils.ts

```diff
@@ -0,0 +1,304 @@
+        case JobListStateActionTypes.view:
+            return { ...state, view: action.payload }
+        case JobListStateActionTypes.code:
+            return { ...state, code: action.payload }
+        case JobListStateActionTypes.errors:
+            return { ...state, errors: action.payload }
+        case JobListStateActionTypes.jobs:
+            return { ...state, jobs: action.payload }
+        case JobListStateActionTypes.size:
+            return { ...state, size: action.payload }
+        case JobListStateActionTypes.sortRule:
+            return { ...state, sortRule: action.payload }
+        case JobListStateActionTypes.showCommandBar:
+            return { ...state, showCommandBar: action.payload }
+        case JobListStateActionTypes.offset:
+            return { ...state, offset: action.payload }
+        case JobListStateActionTypes.pageSize:
+            return { ...state, pageSize: action.payload }
+        case JobListStateActionTypes.expandedRow:
+            return { ...state, expandedRow: action.payload }
+        case JobListStateActionTypes.isAllExpanded:
+            return { ...state, isAllExpanded: action.payload }
+        case JobListStateActionTypes.isAllExpandable:
+            return { ...state, isAllExpandable: action.payload }
+        case JobListStateActionTypes.multipleOptions:
+            return { ...state, ...action.payload }
+        default:
+            return state
+    }
+}
+
+export const jobListModal = (jobContainers) => {
+    return (
+        jobContainers?.map((job) => {
+            return {
+                id: job.jobId || 0,
+                name: job.jobName || 'NA',
+                description: job.description || '',
+                ciPipelines: pipelineModal(job.ciPipelines),
+                defaultPipeline: getDefaultPipeline(job.ciPipelines),
+            }
+        }) ?? []
+    )
+}
+
+const pipelineModal = (ciPipelines: JobCIPipeline[]) => {
+    return (
+        ciPipelines?.map((ciPipeline) => {
+            let status = ciPipeline.status
+            if (ciPipeline.status.toLocaleLowerCase() == 'deployment initiated') {
+                status = 'Progressing'
+            }
+
+            return {
+                ciPipelineId: ciPipeline.ciPipelineId || 0,
+                ciPipelineName: ciPipeline.ciPipelineName || '',
+                lastRunAt: ciPipeline.lastRunAt ? handleUTCTime(ciPipeline.lastRunAt, false) : '',
+                lastSuccessAt: ciPipeline.lastSuccessAt ? handleUTCTime(ciPipeline.lastSuccessAt, false) : '',
+                status: ciPipeline.status ? handleDeploymentInitiatedStatus(ciPipeline.status) : 'notdeployed',
+            }
+        }) ?? []
+    )
+}
+
+const getDefaultPipeline = (ciPipelines) => {
+    if (ciPipelines?.length > 0) {
+        const ciPipeline =
+            ciPipelines.find((pipeline) => pipeline.default) || getLastTriggeredJob(ciPipelines) || ciPipelines[0]
+        let status = ciPipeline.status
```

Comment:
can we extract out 101 to 112 in a separate function so that it can be re used. allso create constants for statuses.